ledger/apply/asset.go

```diff
@@ -140,6 +148,8 @@ func AssetConfig(cc transactions.AssetConfigTxnFields, header transactions.Heade
 
 		delete(record.Assets, cc.ConfigAsset)
 		delete(record.AssetParams, cc.ConfigAsset)
+
+		balances.Deallocate(creator, basics.CreatableIndex(cc.ConfigAsset), basics.AssetCreatable, false)
```

Comment:
I'm not a big fan of the usage of the `basics.CreatableLocator` above, since it promotes the `missed field initialization` type-bug. I'd rather seeing the object being constructed as a result of `makeCreatableLocator(..)`

Reply:
could you elaborate please? not sure I follow the point of using `makeCreatableLocator` here

Reply:
sure;
How can you tell that you did not forget to initialize one of the fields in the `deleted` object ?
Creating a wrapper function around that would ensure that you won't be forgetting to initialize all the field members, even when you add a new field to the structure.

Reply:
Ok, I'll submit a separate PR to master with this fix since it is not related to this PR.